### PR TITLE
Add wait plugin

### PR DESCRIPTION
**Pull request by sandrobonazzola**
## Changes introduced with this PR

Add wait plugin from https://github.com/sandrobonazzola/arcaflow-plugin-wait development repo.

---
By contributing to this repository, I agree to the [contribution guidelines](https://github.com/arcalot/.github/blob/main/CONTRIBUTING.md).

Signed-off-by: Sandro Bonazzola <sbonazzo@redhat.com>

### CODE REVIEW

File: python/wait/requirements.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+git+https://github.com/arcalot/arcaflow-plugin-sdk-python.git@0e878bbe60b2a4e8e22532064e88c750e990ef8b
```
Comment (mkarg75) [inline python/wait/requirements.txt:1]: is there a reason to point to a specific commit?

Comment (sandrobonazzola) [reply python/wait/requirements.txt:1]: in absence of a tag, just taking a version which has been tested with the plugin. Will be replaced with standard latest pip release as a new release with included fixes for float validation will be available.

File: python/wait/wait_plugin.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    """
+    This is the output data structure for the success case.
+    """
+    message: str
+
+
+@dataclass
+class ErrorOutput:
+    """
+    This is the output data structure in the error  case.
```
Comment (sanjacodes) [inline python/wait/wait_plugin.py:34]: Double whitespace

File: python/wait/wait_plugin.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    outputs={"success": SuccessOutput, "error": ErrorOutput},
+)
+def wait(
+    params: InputParams
+) -> typing.Tuple[str, typing.Union[SuccessOutput, ErrorOutput]]:
+    """
+    :param params:
+
+    :return: the string identifying which output it is,
+             as well the output structure
```
Comment (sanjacodes) [inline python/wait/wait_plugin.py:52]: as well as?

**Review by sanjacodes — APPROVED** [review]

File: python/wait/wait_plugin.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    """
+    try:
+        sleep(params.seconds)
+        return "success", SuccessOutput(
+            "Waited for {} seconds".format(params.seconds)
```
Comment (jaredoconnell) [inline python/wait/wait_plugin.py:57]: Do we want to assume that the waiting time was the sleep time? The alternative option is to add two time.time() calls to get the actual sleep time.

File: python/wait/wait_plugin.py (hunk 2 above)
Comment (sandrobonazzola) [reply python/wait/wait_plugin.py:34]: done

File: python/wait/requirements.txt (hunk 1 above)
Comment (sandrobonazzola) [reply python/wait/requirements.txt:1]: ok

File: python/wait/wait_plugin.py (hunk 3 above)
Comment (sandrobonazzola) [reply python/wait/wait_plugin.py:52]: done

**Comment by dustinblack** [comment]
There are always places where a true timed sleep is an unavoidable need, but I hope this plugin doesn't encourage laziness in workflow development. Optimally every pause in a workflow will have the context for what it's waiting for and will use an appropriate state check instead of a timed sleep.

**Comment by sandrobonazzola** [comment]
Abandoning this PR as we are splitting to individual repository for each plugin.
Going to transfer https://github.com/sandrobonazzola/arcaflow-plugin-wait to arcalot org.

**Pull request closed without merging**